tcms/report/views.py

```diff
@@ -81,71 +82,83 @@ def overview(request, product_id, template_name='report/overview.html'):
-            version.cases_count = cases_subtotal.get(vid, 0)
-            version.failed_case_runs_count = failed_case_runs_subtotal.get(vid,
-                                                                           0)
-
-            m = finished_case_runs_subtotal.get(vid, 0)
-            n = case_runs_subtotal.get(vid, 0)
-            if m and n:
-                version.case_run_percent = round(m * 100.0 / n, 1)
-            else:
-                version.case_run_percent = .0
+        product_id = self.__class__.product.pk
 
         case_runs_status_subtotal = None
         selected_version = getattr(self, 'selected_version', None)
+
         if selected_version is not None:
             case_runs_status_subtotal = self.case_runs_status_subtotal(
                 product_id, selected_version.pk)
 
         data = super(ProductVersionReport, self).get_context_data(**kwargs)
+
         data.update({
             'SUB_MODULE_NAME': 'version',
-            'product': self.product,
-            'versions': versions,
+            'product': self.__class__.product,
+            'versions': self._get_updated_versions(product_id),
             'version': selected_version,
             'case_runs_status_subtotal': case_runs_status_subtotal,
         })
 
         return data
 
+    def _get_updated_versions(self, product_id):
+        versions = self.__class__.product.version.only('product', 'value')
+
+        plans_subtotal = self.plans_subtotal(product_id)
+        running_runs_subtotal = self.runs_subtotal(product_id, True)
+        finished_runs_subtotal = self.runs_subtotal(product_id, False)
+        cases_subtotal = self.cases_subtotal(product_id)
+        case_runs_subtotal = self.case_runs_subtotal(product_id)
+        finished_case_runs_subtotal = self.finished_case_runs_subtotal(product_id)
+        failed_case_runs_subtotal = self.failed_case_runs_subtotal(product_id)
+
+        for version in versions:
+            version_id = version.pk
+
+            version.plans_count = plans_subtotal.get(version_id, 0)
+            version.running_runs_count = running_runs_subtotal.get(version_id, 0)
+            version.finished_runs_count = finished_runs_subtotal.get(version_id, 0)
+            version.cases_count = cases_subtotal.get(version_id, 0)
+            version.failed_case_runs_count = failed_case_runs_subtotal.get(version_id, 0)
+
+            subtotal_finished_runs = finished_case_runs_subtotal.get(version_id, 0)
+            subtotal_runs = case_runs_subtotal.get(version_id, 0)
+
+            if subtotal_finished_runs and subtotal_runs:
+                version.case_run_percent = round(subtotal_finished_runs * 100.0 / subtotal_runs, 1)
+            else:
+                version.case_run_percent = .0
+
+        return versions
+
 
 class ProductBuildReport(TemplateView, ProductBuildReportData):
     template_name = 'report/build.html'
 
-    def get(self, request, product_id):
+    def __init__(self):
+        self.product = None
```

Comment:
Move to class attributes like you did for the other class. Do this for all classes where you have this pattern.